polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1546 @@
+                continue
+            elif processing_type == 'cumsum':
+                variables[var]['path_values'] = np.cumsum(variables[var]['path_values'])
+
+
+        path_info = {}
+        path_info['path']      = np.array(path_points)
+        path_info['variables'] = variables
+
+        return path_info
+
+#======================================================
+class Smoothing:
+    def __init__(self,dijkstra_graph,adjacent_pairs,start_waypoint,end_waypoint,blocked_metric='SIC'):
+        '''
+            Class construct that has all the operations requried for path smoothing. Including: Relationship of adjacent pairs,
+            edge finding new edges to add and returns a list of the adjacent pairs for the constructed path
+
+        '''
+        self._initialise_config()
+        self.dijkstra_graph = dijkstra_graph
+        self.aps = adjacent_pairs
+        self.start_waypoint = start_waypoint
+        self.end_waypoint = end_waypoint
+        self.blocked_metric = blocked_metric
+
+
+        for key in self.dijkstra_graph.keys():
+            cell = self.dijkstra_graph[key]
+            if len(cell['neighbourTravelLegs'])>0:
+                accessible_edges = np.where(np.isfinite(np.sum(cell['neighbourTravelLegs'],axis=1)))[0]
+                cell['case'] = cell['case'][accessible_edges]
+                cell['neighbourIndex'] = cell['neighbourIndex'][accessible_edges]
+                cell['neighbourCrossingPoints'] = cell['neighbourCrossingPoints'][accessible_edges]
+                cell['neighbourTravelLegs'] = cell['neighbourTravelLegs'][accessible_edges]
+
+                self.dijkstra_graph[key] = cell
+
+    def _initialise_config(self):
+        '''    
+            Initialising configuration information. If None return a list of standards
+        '''
+
+        self.merge_separation = 1e-3
+        self.converged_sep    = 5e-3
+        self._g = pyproj.Geod(ellps='WGS84')
+
+    def _long_case(self,start,end,case,Sp,Cp,Np):
+        '''
+            Longitude based smoothing updating the crossing point given the conditions 
+            of the adjacency pair
+
+            Input:
+                start (dict) - Dictionary of the start cell information
+                end (dict)   - Dictionary of the end  cell information
+                case (int)   - Adjacency case type connecting the two cells
+                Sp (tuple)   - Start Point (long,lat)
+                Cp (tuple)   - Crossing Point (long,lat)
+                Np (tuple)   - End Point (long,lat)
+
+            Returns:
+                Cp (tuple)   - Updated Crossing Point (long,lat)
+ 
+        '''
+        def NewtonOptimisationLong(f,y0,x,a,Y,u1,v1,u2,v2,speed_s,speed_e,R,λ_s,φ_r):
+                '''
+                    Apply newton optimisation to determine an update to the crossing point.
+                    
+                    All information must be considered in tandem to scientific publication.
```

Comment:
I think it would be worth citing the publication when referring to it in the code, maybe reference one of the old arxiv papers until you have a DOI or something for the new publication?